### ISSUE TITLE
Problem with the Plugin

### ISSUE THREAD

**Issue by TheDodo24** [zDevelopers/ImageOnMap#56]
Hey, my problem is that if I want to render a picture following error message comes:

"[ImageOnMap] Rendering from TheDodo24 failed: java.awt.AWTError: Can't connect to X11 window server using 'localhost:12.0' as the value of the DISPLAY variable."

Can you please help me?

**Comment by AmauryCarrade** [comment]
I don't have an answer right now (need to investigate) but, what is your Java version, system specs, Bukkit/Minecraft version?

**Comment by TheDodo24** [comment]
Java Version: 1.8.0_191
For the server we use Paperspigot 1.12.2-R0.1-SNAPSHOT
As server we have Debian 9 on 8 CPU Cores and 24GB RAM

**Comment by TheDodo24** [comment]
Now I have the error message: 

[ImageOnMap] Rendering from TheDodo24 failed: java.lang.NoClassDefFoundError: Could not initialize class sun.awt.X11GraphicsEnvironment

**Comment by totemo** [comment]
Concerned citizen here. :)

This bug looks like ancient history and it's a duplicate of #52, so it should be closed. I recommend adding the following information to the README or other documentation:

 * If you are running a server on a Linux system (e.g. a virtual private server) you probably don't have display, mouse and keyboard devices attached. You should therefore ensure that the `$DISPLAY` environment variable is unset. Use `unset DISPLAY` in the shell before starting the server.
* The high number in `$DISPLAY` (12 here) is usually symptomatic of logging in with SSH using X11 forwarding (i.e. `ssh -X` or `ssh -Y`). I most cases it is unnecessary to do so, because you are only using the command line.
 * Background reading:
   * https://www.oracle.com/technical-resources/articles/javase/headless.html
   * https://stackoverflow.com/questions/41067235/what-is-the-benefit-of-setting-java-awt-headless-true

Per that documentation, defining `-Djava.awt.headless=true` should be sufficient for Java's graphics APIs to degrade to a headless mode of operation. What you are seeing is probably some quirk of the version of Java you're using, or perhaps you put `-Djava.awt.headless=true` after the JAR file name by mistake (in which case it would have no effect).

On Fedora Linux, with PaperSpigot 1.15.2 build 128 and openjdk version "13.0.2" 2020-01-14, I can confirm that `export DISPLAY=:12.0` makes the plugin (actually Java AWT) shit itself (as expected) but also, `-Djava.awt.headless=true` fixes it (also, as expected).

In my test, running `unset DISPLAY` before starting the server restores normality.

**Comment by AmauryCarrade** [comment]
Thanks you for your tips! I'll add them into the documentation as it's indeed something that can happen frequently.
